src/dsymbol/tests.d

```diff
@@ -298,6 +298,38 @@ unittest
 	assert(T2.kind == CompletionKind.variadicTmpParam);
 }
 
+// check for memory leaks on thread termination (in static constructors)
+version (linux)
+unittest
+{
+	import core.memory : GC;
+	import core.thread : Thread;
+	import fs = std.file;
+	import std.array : split;
+	import std.conv : to;
+
+	// create and destroy a lot of dummy threads
+	foreach (j; 0 .. 100)
+	{
+		Thread[100] arr;
+		foreach (i; 0 .. 100)
+			arr[i] = new Thread({}).start();
+		foreach (i; 0 .. 100)
+			arr[i].join();
+	}
+
+	GC.collect();
+	GC.minimize();
+
+	// get Linux process statistics
+	const txt = fs.readText("/proc/self/stat");
```

Comment:
this should probably read before creating any threads and compare the difference to be more accurate

Reply:
Yeah, it's better, but not much. I made it x2 more accurate and x2 faster. GC or runtime keeps some memory for threads even after minimization.

Reply:
so how many pages does it roughly take now?

Reply:
The difference is 2-3k pages now. If I warm up the GC by creating and destroying 100 threads before the test, it is 0.5-1k. It fluctuates between tests.
The more threads, the more precision. I can see if there is any small leak if I increase the thread amount to ~100k, but it's too long.